idx/cassandra/cassandra.go

```diff
@@ -315,8 +315,12 @@ func (c *CasIdx) updateCassandra(now uint32, inMemory bool, archive idx.Archive,
 func (c *CasIdx) rebuildIndex() {
 	log.Info("cassandra-idx: Rebuilding Memory Index from metricDefinitions in Cassandra")
 	pre := time.Now()
-	defs := c.LoadPartitions(cluster.Manager.GetPartitions(), nil, pre)
-	num := c.MemoryIdx.Load(defs)
+	var defs []schema.MetricDefinition
+	var num int
+	for _, partition := range cluster.Manager.GetPartitions() {
+		defs = c.LoadPartitions([]int32{partition}, defs[:0], pre)
```

Comment:
@woodsaj isn't this a bug? subsequent calls of this seem to overwrite previously loaded defs?

Reply:
no.